advanced/button-interrupt/solution/src/main.rs

```diff
@@ -12,30 +12,15 @@ use esp_idf_sys::{
 
```

Comment:
cargo-clippy spotted a few things. Actually we should have a pipeline that runs all the solutions through clippy.

```
warning: unused imports: `ffi::CString`, `sync::mpsc`
 --> src/main.rs:4:11
  |
4 | use std::{ffi::CString, ptr, sync::mpsc};
  |           ^^^^^^^^^^^^       ^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: unused import: `xTaskCreatePinnedToCore`
  --> src/main.rs:10:60
   |
10 |     xQueueGenericCreate, xQueueGiveFromISR, xQueueReceive, xTaskCreatePinnedToCore, QueueHandle_t,
   |                                                            ^^^^^^^^^^^^^^^^^^^^^^^

warning: unreachable pattern
  --> src/main.rs:70:17
   |
69 |                 _ => println!("button pressed!"),
   |                 - matches any value
70 |                 0 => {}
   |                 ^ unreachable pattern
   |
   = note: `#[warn(unreachable_patterns)]` on by default
```